SofaKernel/modules/SofaHelper/SofaHelper_test/narrow_cast_test.cpp

```diff
@@ -0,0 +1,71 @@
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <sofa/helper/narrow_cast.h>
+#include <gtest/gtest.h>
+
+namespace sofa
+{
+    TEST(narrow_cast_test, in_range)
+    {
+        {
+            size_t a = 0;
+            int b = 0;
+            EXPECT_EQ(sofa::helper::narrow_cast<int>(a), b);
+        }
+
+        {
+            const size_t a = 0;
+            int b = 0;
+            EXPECT_EQ(sofa::helper::narrow_cast<int>(a), b);
+        }
+
+        {
+            const size_t a = 0;
+            const int b = 0;
+            EXPECT_EQ(sofa::helper::narrow_cast<int>(a), b);
+        }
+
+        {
+            constexpr size_t a = std::numeric_limits<int>::max();
+            constexpr int b = std::numeric_limits<int>::max();
+            EXPECT_EQ(sofa::helper::narrow_cast<int>(a), b);
+        }
+
+    }
+
+    // Situations where narrow conversion changes the input value
+    TEST(narrow_cast_test, out_of_range)
+    {
+        {
+            constexpr size_t a = static_cast<size_t>(std::numeric_limits<int>::max()) + 1;
+            if constexpr (sofa::helper::forceNarrowCastChecking)
+            {
+                EXPECT_THROW(sofa::helper::narrow_cast<int>(a), sofa::helper::narrowing_error);
+            }
+            else
+            {
+                const int b = std::numeric_limits<int>::min(); // overflow
+                EXPECT_EQ(sofa::helper::narrow_cast<int>(a), b);
+            }
+        }
+    }
```

Comment:
maybe add 2 tests on casting : 
 - a unsigned int to a int ? 
 - a int to a short or char ?
  (and see throwing error and stuff)

Reply:
done